ironfish/src/rpc/stream.ts

```diff
@@ -36,12 +70,16 @@ export class Stream<T> implements AsyncIterable<T> {
       return Promise.resolve({ done: false, value: value as T })
     }
 
+    if (this.error) {
+      return Promise.reject(this.error)
+    }
```

Comment:
With this, it's possible to read a valid value from a stream that's been ended with an error. Is that okay? That's potentially very strange behaviour that a stream with closed = true can still yield values.

Reply:
Will change this thanks for noticing